x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -670,8 +721,14 @@ export class TaskRunner<
       if (!event.message) {
         event.message = `${this.ruleType.id}:${ruleId}: execution failed`;
       }
+      monitoringHistory.success = false;
     }
 
+    ruleMonitoring.execution.history.push(monitoringHistory);
+    ruleMonitoring.execution.calculated_metrics.success_ratio =
+      ruleMonitoring.execution.history.filter(({ success }) => success).length /
+      ruleMonitoring.execution.history.length;
+    await this.updateMonitoring(ruleId, namespace, ruleMonitoring);
```

Comment:
I have few questions here:

1. do we want to update our success ratio/ count if the rule has been canceled? I think we should do the same thing than the execution status.
2. I am wondering if it will be better to only have one request to update the rule partially and convert `updateRuleExecutionStatus` to `updateRulePartially` and make this function smarter.

what do you think?

Reply:
sure, that sounds good